### PR TITLE
Changelog: Add 1.0 changes

### LINKED ISSUE #1845: Write 1.0 changelog / release notes
In preparation for release, start working on changelog

### PR DESCRIPTION
**Pull request by jku**
Leaving as draft for now:
* I believe I've listed all relevant changes so far (some trivial PRs are not listed) but obviously there are likely to be a few more changes before 1.0 
* The text section could be improved, but I'll do that with a fresher brain

### CODE REVIEW

**Comment by ivanayov** [comment]
Should we add bug fixes?
* [Metadata API: Role keyids is not ordered #1478](https://github.com/theupdateframework/python-tuf/issues/1478) /[role keyids still not ordered #1752](https://github.com/theupdateframework/python-tuf/issues/1752)
* [ngclient: should clean temp file if write fails #1575](https://github.com/theupdateframework/python-tuf/issues/1575)
* Not sure if this was present before 0.20.0: [ngclient: throws securesystemslib errors #1761](https://github.com/theupdateframework/python-tuf/issues/1761)

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,43 @@
...
+* ngclient: Explicitly encode rolename in URL (#1759)
+* ngclient: Allow HTTP payload compression (#1774)
+* ngclient: Make exceptions more consistent (#1799, #1810)
+* docs: Improve documentation (#1744, #1749, #1750, #1755, #1771, #1776, #1772,
+  #1780, #1781, #1800, #1815, #1820, #1829, #1838)
+* build: Various build infrastructure improvements (#1718, #1724, #1760, #1762,
+  #1767, #1803, #1830, #1832, #1837, #1839)
+* build: Stop supporting EOL Python 3.6 (#1783)
+* build: Update dependencies (#1827, #1834)
+
```
Comment (ivanayov) [inline docs/CHANGELOG.md:33]: Waiting for latest updates on #1852 as `securesystemslib` updates are already present.

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,43 @@
...
+## v1.0.0
+
+This release makes ngclient and the Metadata API the supported python-tuf APIs.
+It also removes the legacy implementation as documented in the
+[1.0.0 announcement](1.0.0-ANNOUNCEMENT.md): all library code is now contained
+in `tuf.api` or `tuf.ngclient`.
+
+### Added
+* tests: Extend client testing (#1689, #1703, #1711, #1728, #1735, #1738,
+  #1742, #1766, #1777, 1831)
```
Comment (ivanayov) [inline docs/CHANGELOG.md]: Missing hash sign: #1831

**Review by ivanayov — COMMENTED** [review]
Seems like almost everything is present. Leaving some minor comments.

Should we add bug fixes?
* [Metadata API: Role keyids is not ordered #1478](https://github.com/theupdateframework/python-tuf/issues/1478) /[role keyids still not ordered #1752](https://github.com/theupdateframework/python-tuf/issues/1752)
* [ngclient: should clean temp file if write fails #1575](https://github.com/theupdateframework/python-tuf/issues/1575)
* Not sure if this was present before 0.20.0: [ngclient: throws securesystemslib errors #1761](https://github.com/theupdateframework/python-tuf/issues/1761)

**Comment by jku** [comment]
> Should we add bug fixes?

They should be in the list, just with the PR number and not the issue number (I can see an argument for using issue numbers: it's just potentially a lot more work since `git log --merges v0.20.0..origin/develop` gives a good initial list of PRs, but a complete list of issues is much harder to collect). 

Out of your list all others have PRs in the list except 1754 and 1478: the latter was "fixed" last summer already but the former does seem to have the wrong issue number in the changelog. Thanks!

I fixed that and rebased to include the latest PRs 1850, 1848, 1855, 1809.

**Comment by ivanayov** [comment]
It looks good this way as well (i.e. without specific "Bug fixes" section).

I think that after having all latest required PRs merged (including the not yet opened for #1852) this should be good to go.

**Comment by jku** [comment]
rebased, added a commit that documents the new PRs in that rebase (this can be squashed, just trying to make this easier to review)

**Comment by ivanayov** [comment]
Looks good to me.

**Comment by jku** [comment]
rebased again, added a commit that documents the new merges in that rebase (this can be squashed, just trying to make this easier to review)

File: docs/CHANGELOG.md (hunk 1 above)
Comment (jku) [reply docs/CHANGELOG.md:33]: this was added as 1861, thanks

**Review by MVrachev — APPROVED** [review]
I had a look at all commits after release `0.20.0` and it seems that all changes are here.
LGTM!

**Comment by lukpueh** [comment]
Superseded by #1873

**Pull request closed without merging**